Fix/mailmap too many values unpack only

yarikoptic opened the pull request:
An enhanced part (no blacking, only relevant changes + orcid sample fixup) of #61 . Fixes #60

vsoch committed:
fixing mailmap issue

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

vsoch committed:
updating test with orcid id that has an "other names"

Signed-off-by: vsoch <vsoch@users.noreply.github.com>

Rebased by debian@onerussian.com

yarikoptic committed:
BF+ENH(TST): first name in the line should "win" in mailmap

- also IMHO simplified code/logic
- added basic testing for parsing .mailmap

yarikoptic committed:
RF(TST): move mailmap tests into centralized src tree tests/

vsoch commented on tributors/main/parsers/mailmap.py:
```diff
@@ -1,6 +1,6 @@
 """
 
-Copyright (C) 2020 Vanessa Sochat.
+Copyright (C) 2020-2021 Vanessa Sochat.
```
```suggestion
Copyright (C) 2020-2022 Vanessa Sochat.
```

vsoch commented on tributors/main/parsers/mailmap.py:
```diff
@@ -38,9 +38,21 @@ def load_data(self):
+                    # only the first name matters
+                    if not name and new_name:
+                        name = new_name
+                    if not name:
```
Do we really want an error raised here?

yarikoptic replied:
Explicit better than implicit... Badly formatted file better leads to error than wrong result IMHO

yarikoptic committed:
Update to this year

Co-authored-by: Vanessasaurus <814322+vsoch@users.noreply.github.com>

vsoch commented:
This looks good to me! I don't use mailmap so I think you are likely the main audience for this. This is ready to merge, but could you bump the version and add a note to the changelog?

Thanks for working on this!

yarikoptic committed:
boost version and changelog for 0.0.22

Filed also https://github.com/con/tributors/issues/73
to avoid such manual dances
